formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/conversion/StmtPureConversionVisitor.kt

```diff
@@ -0,0 +1,92 @@
+package org.jetbrains.kotlin.formver.core.conversion
+
+import org.jetbrains.kotlin.KtSourceElement
+import org.jetbrains.kotlin.fir.FirElement
+import org.jetbrains.kotlin.fir.declarations.FirProperty
+import org.jetbrains.kotlin.fir.expressions.FirBlock
+import org.jetbrains.kotlin.fir.expressions.FirLiteralExpression
+import org.jetbrains.kotlin.fir.expressions.FirPropertyAccessExpression
+import org.jetbrains.kotlin.fir.expressions.FirReturnExpression
+import org.jetbrains.kotlin.fir.expressions.impl.FirUnitExpression
+import org.jetbrains.kotlin.fir.visitors.FirVisitor
+import org.jetbrains.kotlin.formver.common.SnaktInternalException
+import org.jetbrains.kotlin.formver.common.UnsupportedFeatureBehaviour
+import org.jetbrains.kotlin.formver.core.embeddings.expression.*
+import org.jetbrains.kotlin.text
+import org.jetbrains.kotlin.types.ConstantValueKind
+
+/**
+ * A restrictive version of the StmtConversionVisitor, which is used to embed functions annotated as pure
+ * into an ExpEmbedding representation that reflects their pure aspects better and makes linearization easier
+ */
+object StmtPureConversionVisitor : FirVisitor<ExpEmbedding, StmtConversionContext>() {
+    override fun visitElement(element: FirElement, data: StmtConversionContext): ExpEmbedding =
+        handleUnimplementedElement(
+            element.source,
+            "Pure conversion not yet implemented for $element (${element.source.text}) - Consider removing the @Pure annotation of this function",
+            data
+        )
+
+    override fun visitReturnExpression(
+        returnExpression: FirReturnExpression,
+        data: StmtConversionContext,
+    ): ExpEmbedding {
+        val expr = when (returnExpression.result) {
+            is FirUnitExpression -> NullLit
```

Comment:
This is wrong: there is (or should be) an object of type `Unit`.  If it's hard to get to work, throw an exception.